### PR TITLE
Feature/252:Add Mothers Name under visits overview page

### PR DESCRIPTION
**Pull request by sherrif10**
PR  https://sd-vxnaid.atlassian.net/browse/VRVU-252   cc @druchniewicz  @jdolkowski

### CODE REVIEW

File: app/src/main/java/com/jnj/vaccinetracker/common/data/models/Constants.kt [hunk 1]
```diff
@@ -108,6 +108,7 @@ object Constants {
     const val CLIENT_ID_FILE_COLUMN_HEADER = "Client ID"
     const val CLIENT_NAME_FILE_COLUMN_HEADER = "Client Name"
     const val PHONE_NUMBER_FILE_COLUMN_HEADER = "Phone Number"
+    const val CLIENT_MOTHERFULL_COLUMN_NAME_HEADER = "Mother Full Name"
```
Comment (druchniewicz) [inline app/src/main/java/com/jnj/vaccinetracker/common/data/models/Constants.kt]: I'd change it just to Mother Name

File: app/src/main/java/com/jnj/vaccinetracker/common/domain/entities/Participant.kt [hunk 2]
```diff
@@ -32,6 +32,7 @@ sealed class ParticipantBase {
     val motherLastName: String? get() = attributes[Constants.ATTRIBUTE_MOTHER_LAST_NAME]
     val childCategory: String? get() = attributes[Constants.ATTRIBUTE_CHILD_CATEGORY]
     val fullName: String get() = listOfNotNull(childFirstName, childLastName).joinToString(" ")
+    val motherFullName: String get() = listOfNotNull(motherFirstName,motherLastName).joinToString { "" }
```
Comment (druchniewicz) [inline app/src/main/java/com/jnj/vaccinetracker/common/domain/entities/Participant.kt]: 1. add space after "motherFirstName"
2. you want to combine these names and separate by " " so it is wrong. Take a look line above how it should be done

File: app/src/main/java/com/jnj/vaccinetracker/visitsoverview/dialog/VisitDetailsDialog.kt [hunk 3]
```diff
@@ -45,7 +45,8 @@ class VisitDetailsDialog : BaseDialogFragment() {
             binding.tableRowVaccines,
             binding.tableRowPhoneNumber,
             binding.tableRowClientId,
-            binding.tableRowClientName
+            binding.tableRowClientName,
+            binding.tableRowClientMotherFullName
```
Comment (druchniewicz) [inline app/src/main/java/com/jnj/vaccinetracker/visitsoverview/dialog/VisitDetailsDialog.kt:49]: please change to tableRowClientMotherName

File: app/src/main/java/com/jnj/vaccinetracker/visitsoverview/dto/VisitDetailsDTO.kt [hunk 4]
```diff
@@ -9,5 +9,6 @@ class VisitDetailsDTO(
     val visitType: String,
     val phoneNumber: String,
     val clientID: String,
-    val clientFullName: String
+    val clientFullName: String,
+    val clientMotherFullName: String
```
Comment (druchniewicz) [inline app/src/main/java/com/jnj/vaccinetracker/visitsoverview/dto/VisitDetailsDTO.kt]: change to clientMotherName

File: app/src/main/java/com/jnj/vaccinetracker/common/data/models/Constants.kt (hunk 1 above)
Comment (druchniewicz) [reply app/src/main/java/com/jnj/vaccinetracker/common/data/models/Constants.kt]: and please also change const name of course

File: app/src/main/java/com/jnj/vaccinetracker/visitsoverview/screens/VisitsListFragment.kt [hunk 5]
```diff
@@ -164,20 +164,22 @@ class VisitsListFragment(private val visitsKey: String) : BaseFragment(),
             headerRow.createCell(1).setCellValue(Constants.CLIENT_ID_FILE_COLUMN_HEADER)
             headerRow.createCell(2).setCellValue(Constants.CLIENT_NAME_FILE_COLUMN_HEADER)
             headerRow.createCell(3).setCellValue(Constants.PHONE_NUMBER_FILE_COLUMN_HEADER)
+            headerRow.createCell(3).setCellValue(Constants.CLIENT_MOTHERFULL_COLUMN_NAME_HEADER)
```
Comment (druchniewicz) [inline app/src/main/java/com/jnj/vaccinetracker/visitsoverview/screens/VisitsListFragment.kt]: wrong index in createCell

File: app/src/main/java/com/jnj/vaccinetracker/visitsoverview/screens/VisitsListFragment.kt [hunk 6]
```diff
@@ -164,20 +164,22 @@ class VisitsListFragment(private val visitsKey: String) : BaseFragment(),
...
                 row.createCell(3).setCellValue(visit.participant.phone)
+                row.createCell(4).setCellValue(visit.participant.motherFullName)
             }
 
             sheet.setColumnWidth(0, 4000)
             sheet.setColumnWidth(1, 4000)
             sheet.setColumnWidth(2, 7000)
             sheet.setColumnWidth(3, 4000)
-
+            sheet.setColumnWidth(4, 4000)
```
Comment (druchniewicz) [inline app/src/main/java/com/jnj/vaccinetracker/visitsoverview/screens/VisitsListFragment.kt]: remove this empty line. please be consistent. 
and I'd increase width of the column because mother full name can long. so please change it to 7000

File: app/src/main/res/layout/item_visit_record.xml [hunk 7]
```diff
@@ -53,7 +53,20 @@
...
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:text="@{visit.participant.motherFullName}"
+            android:textSize="16sp"
+            android:textStyle="bold"
+            android:gravity="center"
+            app:layout_constraintStart_toEndOf="@id/tv_clientName"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintWidth_percent="0.2" />
```
Comment (druchniewicz) [inline app/src/main/res/layout/item_visit_record.xml]: this has 0.2 width, another TextViews have 0.25. Please keep the same width and smallest one for eye icon e.g.
0.21
0.21
0.21
0.21
0.16

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -84,13 +84,14 @@
...
     <string name="participant_matching_title">Child matching</string>
+    <string name="visits_overview_client_mothers_name_column_label">Mother's Name</string>
     <string name="participant_flow_child_number_title">Child Number</string>
     <string name="participant_flow_participant_id_title">Child ID</string>
     <string name="participant_flow_participant_id_hint">Scan Child ID QR Code</string>
     <string name="participant_flow_participant_id_scanned">The format of the ID is not supported</string>
     <string name="participant_flow_participant_id_description">Please scan the Child vaccination card with a QR-Code or manually enter the Child ID number:</string>
-    <string name="participant_flow_participant_id_label_manual_entry">Manual ID Entry</string>
-    <string name="participant_flow_participant_id_input_hint">#1234567890</string>
+    <string name="participant_flow_participant_id_label_manual_entry">Manual Enter ID or Search by mothers Name</string>
```
Comment (druchniewicz) [inline app/src/main/res/values/strings.xml]: I dont understand why you changed this. These labels are probably used in the Child matching screen where you can look for a child so why did you change it?

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -213,7 +214,7 @@
     <string name="visit_dosing_error_no_vial_barcode">Please enter or scan a vial barcode</string>
     <string name="visit_dosing_error_no_manufacturer">Please select a manufacturer</string>
     <string name="visit_dosing_label_success">Visit successfully registered!</string>
-    <string name="visit_schedule_next_visit_title">Schedule next visit</string>
+    <string name="visit_schedule_next_visit_title">Save visit</string>
```
Comment (druchniewicz) [inline app/src/main/res/values/strings.xml:217]: is it requirement from stakeholders? In my opinion 'Schedule next visit' more precisely says that you're scheduling a next visit

**Review by druchniewicz — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml (hunk 9 above)
Comment (sherrif10) [reply app/src/main/res/values/strings.xml:217]: This is a requirement .

File: app/src/main/res/values/strings.xml (hunk 8 above)
Comment (sherrif10) [reply app/src/main/res/values/strings.xml]: This is a requirement from stakeholders  Eva and the MOH team brought it up.

Comment (druchniewicz) [reply app/src/main/res/values/strings.xml]: ok, but it is not implemented yet so if you change label now it will be confusing that user can search by mother name but in reality he can't because it is not implemented

File: app/src/main/java/com/jnj/vaccinetracker/common/domain/entities/Participant.kt [hunk 10]
```diff
@@ -32,6 +32,7 @@ sealed class ParticipantBase {
     val motherLastName: String? get() = attributes[Constants.ATTRIBUTE_MOTHER_LAST_NAME]
     val childCategory: String? get() = attributes[Constants.ATTRIBUTE_CHILD_CATEGORY]
     val fullName: String get() = listOfNotNull(childFirstName, childLastName).joinToString(" ")
+    val motherName: String get() = listOfNotNull(motherFirstName, motherLastName).joinToString { "" }
```
Comment (druchniewicz) [inline app/src/main/java/com/jnj/vaccinetracker/common/domain/entities/Participant.kt]: I've already mentioned about in previous code review. Please use correct joinToString syntax, look at the line above.

File: app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -84,12 +84,13 @@
     <string name="iris_scan_left_title">Iris scan - Left eye</string>
     <string name="iris_scan_right_title">Iris scan - Right eye</string>
     <string name="participant_matching_title">Child matching</string>
+    <string name="visits_overview_client_mothers_name_column_label" translatable="false">Mother's Name</string>
```
Comment (druchniewicz) [inline app/src/main/res/values/strings.xml]: why it's not translatable?

**Review by druchniewicz — CHANGES_REQUESTED** [review]
